mud-contracts/world/src/modules/smart-gate/systems/SmartGate.sol

```diff
@@ -0,0 +1,221 @@
+   * @notice Unlink Smart Gates
+   * @param sourceGateId is the id of the source gate
+   * @param destinationGateId is the id of the destination gate
+   */
+  function unlinkSmartGates(uint256 sourceGateId, uint256 destinationGateId) public {
+    //Check if the gates are linked
+    if (!isGateLinked(sourceGateId, destinationGateId)) {
+      revert SmartGate_GateNotLinked(sourceGateId, destinationGateId);
+    }
+    SmartGateLinkTable.set(_namespace().smartGateLinkTableId(), sourceGateId, destinationGateId, false);
+  }
+
+  /**
+   * @notice Configure Smart Turret
+   * @param smartGateId is smart object id of the Smart Turret
+   * @param systemId is the system id of the Smart Turret logic
+   */
+  function configureSmartGate(
+    uint256 smartGateId,
+    ResourceId systemId
+  ) public onlyAdminOrObjectOwner(smartGateId) hookable(smartGateId, _systemId()) {
+    SmartGateConfigTable.set(_namespace().smartGateConfigTableId(), smartGateId, systemId);
+  }
+
+  /**
+   * @notice view function for smart gates which is linked
+   * @param characterId is of the id of the character
+   * @param sourceGateId is the id of the source gate
+   * @param destinationGateId is the id of the destination gate
+   */
+  function canJump(uint256 characterId, uint256 sourceGateId, uint256 destinationGateId) public returns (bool) {
+    State sourceGateState = DeployableState.getCurrentState(_namespace().deployableStateTableId(), sourceGateId);
+
+    State destinationGateState = DeployableState.getCurrentState(
+      _namespace().deployableStateTableId(),
+      destinationGateId
+    );
+
+    if (sourceGateState != State.ONLINE) {
+      revert SmartDeployableErrors.SmartDeployable_IncorrectState(sourceGateId, sourceGateState);
+    }
+
+    if (destinationGateState != State.ONLINE) {
+      revert SmartDeployableErrors.SmartDeployable_IncorrectState(destinationGateId, destinationGateState);
+    }
+
+    //Check if the gates are linked
+    if (!isGateLinked(sourceGateId, destinationGateId)) {
+      revert SmartGate_GateNotLinked(sourceGateId, destinationGateId);
+    }
+
+    ResourceId systemId = SmartGateConfigTable.get(_namespace().smartGateConfigTableId(), sourceGateId);
+
+    if (ResourceIds.getExists(systemId)) {
+      bytes memory returnData = world().call(
+        systemId,
+        abi.encodeCall(this.canJump, (characterId, sourceGateId, destinationGateId))
+      );
+      return abi.decode(returnData, (bool));
+    }
+    return true;
+  }
+
+  function isGateLinked(uint256 sourceGateId, uint256 destinationGateId) public view returns (bool) {
+    return (
+      (SmartGateLinkTable.getIsLinked(_namespace().smartGateLinkTableId(), sourceGateId, destinationGateId)) ||
+        (SmartGateLinkTable.getIsLinked(_namespace().smartGateLinkTableId(), destinationGateId, sourceGateId))
+        ? true
+        : false
+    );
+  }
+
+  function calculatDistance(uint256 sourceGateId, uint256 destinationGateId) public view returns (uint256) {
```

Comment:
I guess we can remove it if we don't do it on chain yet

Reply:
This is going to be on-chain in the next PR